edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -74,12 +101,73 @@ await serviceIdentity
                     .Map(s => this.HandleNewServiceIdentity(s))
                     .GetOrElse(() => this.HandleNoServiceIdentity(id));
             }
+            catch (DeviceInvalidStateException)
+            {
+                await this.HandleNoServiceIdentity(id);
+            }
+            catch (Exception e)
+            {
+                Events.ErrorRefreshingCache(e, id);
+            }
+        }
+
+        bool ShouldNotRefresh(StoredServiceIdentity storedServiceIdentity, bool refreshIfOutOfDate) => !refreshIfOutOfDate || storedServiceIdentity.Timestamp + this.refreshDelay > DateTime.UtcNow;
+
+        void VerifyServiceIdentity(StoredServiceIdentity storedServiceIdentity) => this.VerifyServiceIdentity(storedServiceIdentity.ServiceIdentity);
+
+        void VerifyServiceIdentity(Option<ServiceIdentity> serviceIdentity) => serviceIdentity.ForEach(
```

Comment:
Should probably log the status of Verify, if it doesn't get logged anywhere else.